### PR TITLE
Add support for proposed connectivity report

### PR DESCRIPTION
**Pull request by Wazzaps**
Here's the relevant Android app PR with more info: https://invent.kde.org/network/kdeconnect-android/-/merge_requests/236

### CODE REVIEW

**Comment by sredman** [comment]
I can't read the GSConnect code, but just to make sure it's written down... I believe the packet format changed (to accommodate multiple SIM cards) since you originally authored this PR. Make sure to update the PR for the new packet format :smile:

**Comment by Wazzaps** [comment]
> I can't read the GSConnect code, but just to make sure it's written down... I believe the packet format changed (to accommodate multiple SIM cards) since you originally authored this PR. Make sure to update the PR for the new packet format

I did update it, this is the relevant code:

    /**
     * Handle a remote state update.
     *
     * @param {Core.Packet} packet - A kdeconnect.connectivity_report packet
     */
    _receiveState(packet) {
        if (packet.body.signalStrengths) {
            // TODO: Only first SIM (subscriptionID) is supported at the moment
            const subs = Object.keys(packet.body.signalStrengths);
            const firstSub = Math.min.apply(null, subs);
            const data = packet.body.signalStrengths[firstSub];

            this._networkType = data.networkType;
            this._signalStrength = data.signalStrength;
        }

        // Update DBus state
        this.__state.state = this.state;
    }

File: src/service/plugins/connectivity_report.js [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+            return 'network-cellular-3g-symbolic';
+        else if (this.network_type === 'LTE')
+            return 'network-cellular-4g-symbolic';
+        else if (this.network_type === 'EDGE')
+            return 'network-cellular-edge-symbolic';
+        else if (this.network_type === 'GPRS')
+            return 'network-cellular-gprs-symbolic';
+        else if (this.network_type === 'HSPA')
+            return 'network-cellular-hspa-symbolic';
+        else if (this.network_type === '5G') // FIXME: No icon for this!
```
Comment (sonnyp) [inline src/service/plugins/connectivity_report.js:92]: I have opened an issue for this https://gitlab.gnome.org/GNOME/adwaita-icon-theme/-/issues/114

**Review by sonnyp — APPROVED** [review]
Code LGTM 

This is not yet released in KDE Connect Android, so I didn't test but given that you wrote both implementations I figure it's not necessary.

File: src/service/plugins/connectivity_report.js [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+            return 'network-cellular-symbolic';
```
Comment (sonnyp) [inline src/service/plugins/connectivity_report.js:93]:
```suggestion
        else if (this.network_type === '5G') {
            // FIXME: No icon for this!
            // https://gitlab.gnome.org/GNOME/adwaita-icon-theme/-/issues/114
            return 'network-cellular-symbolic';
        }
```

**Pull request merged**